timesketch/frontend-ng/src/components/Explore/TimelineChip.vue

```diff
@@ -104,20 +104,42 @@ limitations under the License.
 
     <v-menu v-else offset-y :close-on-content-click="false" content-class="menu-with-gap" ref="timelineChipMenuRef">
       <template v-slot:activator="{ on }">
-        <v-chip v-on="on" :style="getTimelineStyle(timeline)" class="mr-2 mb-3">
-          <v-icon v-if="timelineStatus === 'fail'" left color="red"> mdi-alert-circle-outline </v-icon>
-          <span class="timeline-name-ellipsis">{{ timeline.name }}</span>
+        <v-chip  @click="toggleTimeline()" :style="getTimelineStyle(timeline)" class="mr-2 mb-3 pr-1 timeline-chip" >
+          <v-icon v-if="timelineFailed" left color="red"> mdi-alert-circle-outline </v-icon>
+          <v-icon v-if="!timelineFailed" left :color="timelineChipColor"> mdi-circle </v-icon>
+
+          <v-tooltip bottom>
+            <template v-slot:activator="{ on: onTooltip, attrs }">
+              <span class="timeline-name-ellipsis" :class="{ disabled: !isSelected && timelineStatus === 'ready'}"
+              v-bind="attrs"
+              v-on="onTooltip"
+              >{{ timeline.name }}</span>
+            </template>
+            <span>{{ timeline.name }}</span>
+          </v-tooltip>
```

Comment:
Is it possible to only show the tooltip when the text is actually truncated?

Reply:
I added a delay of 300ms. It subjectively felt like a good value but we can still change it if you think it is not optimal.

I added a workaround for only showing the tooltip when the text is truncated. The problem is that the truncation is done by css. While there are some hacks that try to detect via JS when the css truncation happens, I added a simple but less precise solution: the tooltip is only shown if the timeline name has more than 30 characters. This does not always match with the css truncation, because characters have different widths.

LMK what you think.

Reply:
Thanks for the changes. 300ms feels very intuitive and the truncation workaround is fine with me as well.